### PR TITLE
Improve benchmarking scripts

### PR DESCRIPTION
**Pull request by juliuscc**

### CODE REVIEW

**Comment by juliuscc** [comment]
This is the output for the new script

Screenshot 2019-11-13 at 09 37 31
Screenshot 2019-11-13 at 09 37 49

File: tools/run_some_benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    print("interval-target:    How much between the ops/sec targets")
+    print("")
+    print("first-threshold:    The lowest speculative retry threshold")
+    print("last-threshold:     The highest speculative retry threshold")
+    print("interval-threshold: How much between the speculative retry thresholds")
+    print("")
+    print("Example:")
+    print("run_some_benchmark.py bench_result 200 800 300 3 5 2")
+    print("This will run benchmarks with the following parameters:")
+    print("200 ops/sec, 3ms threshold")
```
Comment (hannesrabo) [inline tools/run_some_benchmarks.py]: Why are we not including 1/2/4?

File: tools/run_all_benchmarks.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
     exit()
 
 filename_prefix = sys.argv[1]
-operation_targets = np.arange(400, 3600, 200).tolist()
-sr_thresholds = np.arange(1, 20, 1).tolist()
+operation_targets = np.arange(400, 3600 + 1, 200).tolist()
+sr_thresholds = np.arange(1, 20 + 1, 1).tolist()
```
Comment (hannesrabo) [inline tools/run_all_benchmarks.py]: The stop value is non inclusive I suspect?

File: tools/run_some_benchmarks.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    print("500 ops/sec, 3ms threshold")
+    print("800 ops/sec, 3ms threshold")
+    print("200 ops/sec, 5ms threshold")
+    print("500 ops/sec, 5ms threshold")
+    print("800 ops/sec, 5ms threshold")
+    exit()
+
+filename_prefix = sys.argv[1]
+first_target = int(sys.argv[2])
+last_target = int(sys.argv[3]) + 1
```
Comment (hannesrabo) [inline tools/run_some_benchmarks.py]: I think that the +1 would be easier to understand if you added it to the range instead

File: tools/run_some_benchmarks.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+interval_target = int(sys.argv[4])
+first_threshold = int(sys.argv[5])
+last_threshold = int(sys.argv[6]) + 1
+interval_threshold = int(sys.argv[7])
+
+
+operation_targets = np.arange(
```
Comment (hannesrabo) [inline tools/run_some_benchmarks.py]: Why are we using np.range and not just the language feature range()? You are converting it back into a normal python list anyway...

File: tools/run_some_benchmarks.py (hunk 3 above)
Comment (hannesrabo) [reply tools/run_some_benchmarks.py]: I can now see that this was a specific example

File: tools/run_some_benchmarks.py (hunk 1 above)
Comment (hannesrabo) [reply tools/run_some_benchmarks.py]: I can now see that this was a specific example

File: tools/run_all_benchmarks.py (hunk 2 above)
Comment (juliuscc) [reply tools/run_all_benchmarks.py]: Yes

File: tools/run_some_benchmarks.py (hunk 1 above)
Comment (juliuscc) [reply tools/run_some_benchmarks.py]: Yes

File: tools/run_some_benchmarks.py (hunk 4 above)
Comment (juliuscc) [reply tools/run_some_benchmarks.py]: I thought that range did not support a step parameter but I now see that it does. Will fix

File: tools/run_some_benchmarks.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+
+operation_targets = range(first_target, last_target + 1, step_target)
+sr_thresholds = range(first_threshold, last_threshold + 1, step_threshold)
+
+for threshold in sr_thresholds:
+    for target in operation_targets:
+        filename = filename_prefix + "_" + str(threshold) + "_" + str(target)
+        print("Running benchmark for file: " + filename)
+        # os.system("./tools/run_workload.py " + filename +
```
Comment (hannesrabo) [inline tools/run_some_benchmarks.py]: No longer running it?

**Pull request merged**